// @pulumi/tls/privateKey.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateKey = void 0;
class PrivateKey {
    /**
     * Get an existing PrivateKey resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new PrivateKey(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PrivateKey.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateKey.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["algorithm"] = state ? state.algorithm : undefined;
            resourceInputs["ecdsaCurve"] = state ? state.ecdsaCurve : undefined;
            resourceInputs["privateKeyOpenssh"] = state ? state.privateKeyOpenssh : undefined;
            resourceInputs["privateKeyPem"] = state ? state.privateKeyPem : undefined;
            resourceInputs["publicKeyFingerprintMd5"] = state ? state.publicKeyFingerprintMd5 : undefined;
            resourceInputs["publicKeyFingerprintSha256"] = state ? state.publicKeyFingerprintSha256 : undefined;
            resourceInputs["publicKeyOpenssh"] = state ? state.publicKeyOpenssh : undefined;
            resourceInputs["publicKeyPem"] = state ? state.publicKeyPem : undefined;
            resourceInputs["rsaBits"] = state ? state.rsaBits : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.algorithm === undefined) && !opts.urn) {
                throw new Error("Missing required property 'algorithm'");
            }
            resourceInputs["algorithm"] = args ? args.algorithm : undefined;
            resourceInputs["ecdsaCurve"] = args ? args.ecdsaCurve : undefined;
            resourceInputs["rsaBits"] = args ? args.rsaBits : undefined;
            resourceInputs["privateKeyOpenssh"] = undefined /*out*/;
            resourceInputs["privateKeyPem"] = undefined /*out*/;
            resourceInputs["publicKeyFingerprintMd5"] = undefined /*out*/;
            resourceInputs["publicKeyFingerprintSha256"] = undefined /*out*/;
            resourceInputs["publicKeyOpenssh"] = undefined /*out*/;
            resourceInputs["publicKeyPem"] = undefined /*out*/;
        }
    }
}
exports.PrivateKey = PrivateKey;
/** @internal */
PrivateKey.__pulumiType = 'tls:index/privateKey:PrivateKey';
